Library/Homebrew/cmd/bottle.rb

```diff
@@ -183,6 +186,15 @@ def bottle_formula(f)
     relocatable = false
     skip_relocation = false
 
+    formula_source_time = f.stable.stage do
+      newest_mtime = nil
+      Pathname(Dir.pwd).find do |path|
+        newest_mtime ||= path.mtime
+        newest_mtime = path.mtime if path.mtime > newest_mtime
+      end
+      newest_mtime
+    end
```

Comment:
Also FYI, if we support `Tab#write` without timestamp change, we can simplify the logic even further. But I guess you want to consider this in future.